File: Java/Beginner/B_Greedy/Exchange.java
package Beginner.B_Greedy;

import java.util.Scanner;

public class Exchange {
    private static final Scanner scan = new Scanner(System.in);
    private static final int[] coins = { 500, 100, 50, 10 };
    private static int[] coins_cnt = { 0, 0, 0, 0 };

    public static void main(String[] args) {
        int n = scan.nextInt();
        int idx = 0;

        for (int coin : coins) {
            coins_cnt[idx++] += n / coin;
            n %= coin;
        }

        System.out.println("[ 동전의 개수 ]");
        for (int cnt : coins_cnt)
            System.out.print(cnt + " ");

        scan.close();
    }
}

File: v1/Python/beginner/02_Greedy/exchange.py
# 거스름돈 문제
n = int(input())

# 각 동전은 무한대로 존재한다고 가정했을 때, 최소 동전 개수
coins = [500, 100, 50, 10]
coins_cnt = [0, 0, 0, 0]

for i, coin in enumerate(coins):
    coins_cnt[i] += n // coin
    n %= coin 

print("[ 동전의 개수 ]")
print(*coins_cnt)